homeassistant/components/tplink/light.py

```diff
@@ -182,19 +189,23 @@ def __init__(
         device: Device,
         coordinator: TPLinkDataUpdateCoordinator,
         light_module: Light,
+        parent: Device | None = None,
     ) -> None:
         """Initialize the switch."""
         self._light_module = light_module
         # For backwards compat with pyHS100
-        if device.device_type == DeviceType.Dimmer:
+        if device.device_type not in {DeviceType.LightStrip, DeviceType.Bulb}:
             # Dimmers used to use the switch format since
             # pyHS100 treated them as SmartPlug but the old code
             # created them as lights
             # https://github.com/home-assistant/core/blob/2021.9.7/homeassistant/components/tplink/common.py#L86
+            #
+            # With the 0.7 kasa release new device types are added such that the previous check here for dimmers
```

Comment:
Maybe this should be a utility function that returns the unique id ?

Reply:
Now resolved by latest https://github.com/home-assistant/core/pull/117758